### DISCUSSION TITLE
How does dagster embedded elt (sling) discover connections in env.yaml?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18285] [Q&A]
Hi! Can you advise how dagster embedded elt sling discover connections defined in env.yaml and help in building assets. something like a single yaml(env.yaml) representing all the connection details, and can be used for building multi assets as well. code sample for reference:
env.yaml:
connections:
  REDSHIFT:
	type: "redshift"
	host: <host>
	user: <user>
	password: <password>
	port: <port>
	database: <database>
	schema: <schema>
	aws_bucket: <aws_bucket>
	aws_access_key_id: <aws_access_key_id>
	aws_secret_access_key: <aws_secret_access_key>
	
  AWS_S3:
	type: "s3"
	bucket: <bucket_name>
	access_key_id: <access_key_id>
	secret_access_key: <secret_access_key>

  AZURE_STORAGE:
	type: "azure"
	account: <storage_account_name>
	container: <container-name>
	conn_str: <conn_str>

discover the source connection(s3) & target connection(adls) from the YAML, that user specify and build asset with additional path inputs

asset_def = build_sling_asset(
    asset_spec=AssetSpec("my_test_sling_asset"),
    source_stream=<s3_path>,
    target_object=<azure_adls_path>,
    target_options=trg_options,
)

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700856260889089?thread_ts=1700856260.889089&cid=C01U954MEER).*

**Comment by PedramNavid** [comment]
Hi! The current version of dagster-embedded-elt does not currently support reading from a `env.yaml` file but this is something we're planning on adding soon in another release to make it easier to build multiple assets.

**Comment by jvyoralek** [comment]
Hi, I would like to double-check whether this is still planned, as I was unable to find instructions on how to use `.sling/env.yaml` connections within Dagster, which could be really useful.

If I understand correctly, in the current version, I need to redefine all connections from `env.yaml` if I want to use them in Dagster within `SlingResource.connections` array with `SlingConnectionResource` objects right?

```python
sling_resource = SlingResource(
    connections=[
        SlingConnectionResource(
            name="MY_POSTGRES",
            type="postgres",
            connection_string=EnvVar("POSTGRES_CONNECTION_STRING"),
        ),
...
```

Thanks

**Reply by PedramNavid** [reply]
Hi! We ended up not supporting env.yaml, and instead using Dagsters resource system as it’s more flexible. You will need to define the connections in Dagster or possibly write a factory function to create them from the yaml files.

**Reply by jvyoralek** [reply]
Ok, understand, thanks for quick answer.
